resources/views/articles/article.blade.php

```diff
@@ -46,7 +46,7 @@
             <ul class="list-group">
                 @foreach($article->comments()->orderBy('id', 'desc')->get() as $comment)
                     <li class="list-group-item disabled">
-                        <span><span class="text-primary font-weight-bold">{{ $comment->writer->email }}</span> {{ $comment->body }} <i>{{ $comment->timeElapse() }}</i></span>
+                        <span><span class="text-primary font-weight-bold">{{ ($comment->writer->fullName()) ? $comment->writer->username : ' '}}</span> {{ $comment->body }} <i>{{ $comment->timeElapse() }}</i></span>
```

Comment:
This shows username only when it has fullname

1. Should display Fullname
2. When fullname is null/empty, show username
3. when no username, show email address

Hint. 
Just create method `getDisplayName()`  (update `fullName()` method) and put all the logic there and;
in blade, just call `$comment->writer->getDisplayName()`